server/cluster/cluster.go

```diff
@@ -737,7 +742,7 @@ func (c *RaftCluster) processRegionHeartbeat(region *core.RegionInfo) error {
 		regionEventCounter.WithLabelValues("update_cache").Inc()
 	}
 
-	if isNew {
+	if !c.IsPreapred() || isNew {
```

Comment:
Why add `IsPreapred` check here? Just controlling the `isNew` flag should be enough to fix this issue 🤔

Reply:
Yes, but when we have collected enough regions, there is no need to collect the rest anymore.

Reply:
Will this cause the region which is not new actually to be prepared here?

Reply:
Oh, it should be `!c.IsPreapred() && isNew`

Reply:
nice catch!